lib/puppet/file_system/windows.rb

```diff
@@ -133,14 +134,32 @@ def replace_file(path, mode = nil)
              dacl.allow(Puppet::Util::Windows::SID::BuiltinUsers, FILE_READ)
              dacl
            when 0640, 0600
+             dacl = secure_dacl(current_sid)
              secure_dacl(current_sid)
+           when 0660
+             dacl = secure_dacl(current_sid)
+             dacl.allow(Puppet::Util::Windows::SID::BuiltinAdministrators, FILE_WRITE)
+             dacl
+           when 0664
+             dacl = secure_dacl(current_sid)
+             dacl.allow(Puppet::Util::Windows::SID::BuiltinAdministrators, FILE_WRITE)
+             dacl.allow(Puppet::Util::Windows::SID::BuiltinUsers, FILE_READ)
+             dacl
+           when 0666
+             dacl = secure_dacl(current_sid)
+             dacl.allow(Puppet::Util::Windows::SID::BuiltinAdministrators, FILE_WRITE)
+             dacl.allow(Puppet::Util::Windows::SID::BuiltinUsers, FILE_WRITE)
+             dacl
+          when 0444
```

Comment:
you can compact 0444 and 0440 to same when since they both only call `dacl`

Reply:
Done